src/FormattingSettingsService.ts

```diff
@@ -80,33 +81,39 @@ export class FormattingSettingsService implements IFormattingSettingsService {
-                                    ? this.localizationManager.getDisplayName(cardGroupInstance.displayNameKey) : cardGroupInstance.displayName,
-                                description: isSimpleCard ? undefined : (this.localizationManager && cardGroupInstance.descriptionKey)
-                                    ? this.localizationManager.getDisplayName(cardGroupInstance.descriptionKey) : cardGroupInstance.description,
-                                slices: [],
-                                uid: groupUid,
-                                collapsible: cardGroupInstance.collapsible,
-                                delaySaveSlices: cardGroupInstance.delaySaveSlices,
-                                disabled: cardGroupInstance.disabled,
-                                disabledReason: cardGroupInstance.disabledReason,
-                            }
+                            displayName: isSimpleCard ? undefined : (this.localizationManager && cardGroupInstance.displayNameKey)
+                                ? this.localizationManager.getDisplayName(cardGroupInstance.displayNameKey) : cardGroupInstance.displayName,
+                            description: isSimpleCard ? undefined : (this.localizationManager && cardGroupInstance.descriptionKey)
+                                ? this.localizationManager.getDisplayName(cardGroupInstance.descriptionKey) : cardGroupInstance.description,
+                            slices: [],
+                            uid: groupUid,
+                            collapsible: cardGroupInstance.collapsible,
+                            delaySaveSlices: cardGroupInstance.delaySaveSlices,
+                            disabled: cardGroupInstance.disabled,
+                            disabledReason: cardGroupInstance.disabledReason,
+                        }
                         formattingCard.groups.push(formattingGroup);
 
                         // In case formatting model adds data points or top categories (Like when you modify specific visual category color).
                         // these categories use same object name and property name from capabilities and the generated uid will be the same for these formatting categories properties
                         // Solution => Save slice names to modify each slice uid to be unique by adding counter value to the new slice uid
                         const sliceNames: { [name: string]: number } = {};
 
+                        // creating slices using same uid as described above is ok at this stage, powerbi identify the specific slice by the selector.
+                        // so the correct thing to do is to provide a unique selector for each datapoint if needed.
+                        // We are keeping sliceNames for backward compatibility.
+                        // This map helps us to check if we need to change the uid
+                        const selectorsMap: Record<string, { [selector: string]: boolean }> = {}
```

Comment:
this map can be <string, [selector:string]>, In other wrods map of string which is the key to list of strings where its the selectors values list
You can add entry for slice name and its selectors, if a specific selector exist in the list it means it exist, no need for boolean